modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/folder/DLFolderItemSelectorView.java

```diff
@@ -241,6 +243,11 @@ private long _getRepositoryGroupId(long repositoryId) {
 	@Reference
 	private DLAppService _dlAppService;
 
+	@Reference(
+		target = "(model.class.name=com.liferay.portal.kernel.repository.model.Folder)"
+	)
+	private ModelResourcePermission<Folder> _folderModelResourcePermission;
+
 	@Reference
```

Comment:
@W2jx, the target seems be incorrect. It should be https://github.com/liferay-core-infra/liferay-portal/blob/45553a42bea8cd0122d27af03ca78d029b6ab6ab/portal-impl/src/com/liferay/portlet/documentlibrary/service/permission/DLFolderPermission.java#L33

Reply:
Hi @yuhai , The reason why I use `model.class.name=com.liferay.portal.kernel.repository.model.Folder`  is that the method 
```
public static boolean contains(
			PermissionChecker permissionChecker, long groupId, long folderId,
			String actionId)
```
in class DLFolderPermission uses _folderModelResourcePermission.